botorch/generation/gen.py

```diff
@@ -17,6 +17,7 @@
 import numpy as np
 import torch
 from botorch.acquisition import AcquisitionFunction
+from botorch.exceptions.warnings import OptimizationWarning
```

Comment:
There's an issue with botorch warnings getting filtered out in some cases (https://github.com/pytorch/botorch/blob/main/botorch/settings.py#L58). See #1169 for relevant issue & #1170, #1178 for fixes. Let's not use them in critical cases such as this one. `RuntimeWarning` might be a better alternative.